src/unix/config/discover.ml

```diff
@@ -395,9 +395,10 @@ struct
           Configurator.Process.run context
             "opam" ["config"; "var"; "conf-libev:installed"]
         in
-        match result.stdout with
-        | "true\n" -> true
-        | _ -> false
+        match result.stdout, result.exit_code with
+        | "true\n", 0 -> true
+        | _, 0 -> false
+        | _, _ -> raise Not_found
```

Comment:
Could you do one of these two things:

1. Separate the test for the error code into an outer `match` statement, and raise `Failure` instead of `Not_found`, where the payload of `Failure` will explain what it is that failed, as a form of documentation in the source (even though the message will never be used at runtime).

2. (My preference) Keep this combined `match`, but change the second case from `_, 0` to `"false\n", 0`, which will make it clear that the `opam` command doesn't indicate that libev is not installed by returning a non-zero error code, but a non-zero error code means either a more serious failure in `opam`, or the `opam` command is missing. In this case, I'd still suggest to change the exception `Not_found` to `Failure` with a descriptive payload. I found `Not_found` to be misleading, because it first reads like the code means that it is specifically `libev` is not found by opam.

Reply:
For sure, I'll do 2.